app/src/main/java/com/ifttt/groceryexpress/MainActivity.kt

```diff
@@ -1,76 +1,116 @@
 package com.ifttt.groceryexpress
 
-import android.Manifest
 import android.content.Intent
 import android.os.Bundle
+import android.os.PersistableBundle
 import android.text.TextUtils
 import android.view.LayoutInflater
 import android.view.Menu
 import android.view.MenuItem
 import android.view.Window
+import android.widget.LinearLayout
 import android.widget.TextView
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.Toolbar
-import androidx.core.app.ActivityCompat
-import androidx.core.content.PermissionChecker
 import com.google.android.material.textfield.TextInputLayout
 import com.ifttt.connect.ConnectionApiClient
 import com.ifttt.connect.ui.ConnectButton
 import com.ifttt.connect.ui.ConnectResult
 import com.ifttt.connect.CredentialsProvider
 import com.ifttt.groceryexpress.ApiHelper.REDIRECT_URI
 import com.ifttt.location.ConnectLocation
+import com.squareup.picasso.Picasso
 
 class MainActivity : AppCompatActivity() {
 
     private lateinit var connectButton: ConnectButton
     private lateinit var toolbar: Toolbar
     private lateinit var emailPreferencesHelper: EmailPreferencesHelper
+    private lateinit var credentialsProvider: CredentialsProvider
+    private lateinit var connectionId: String
```

Comment:
I don't think we need this as lateinit, since the field is going to be set only after when the user selects an item in the dialog. A nullable var should be enough.

Reply:
the `connectionId` is actually required and connect button cannot be rendered without it, which is why it should not be nullable.